tests/tpch/generate-data.py

```diff
@@ -1,47 +1,145 @@
+import functools
 import pathlib
 import tempfile
+import warnings
 
 import botocore.session
 import click
+import coiled
 import dask
 import duckdb
 import psutil
+import pyarrow.compute as pc
+from dask.distributed import LocalCluster, wait
 
 
 def generate(
-    scale: int = 10, partition_size: str = "50 MiB", path: str = "./tpch-data"
+    scale: int = 10,
+    partition_size: str = "128 MiB",
+    path: str = "./tpch-data",
+    relaxed_schema: bool = False,
 ):
     if str(path).startswith("s3"):
         path += "/" if not path.endswith("/") else ""
-        path += f"scale-{scale}/"
+        path += f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        use_coiled = True
     else:
-        path = pathlib.Path(path) / f"scale-{scale}/"
+        path = (
+            pathlib.Path(path)
+            / f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        )
         path.mkdir(parents=True, exist_ok=True)
+        use_coiled = False
 
     print(f"Scale: {scale}, Path: {path}, Partition Size: {partition_size}")
+    kwargs = dict(
+        scale=scale,
+        path=path,
+        relaxed_schema=relaxed_schema,
+        partition_size=partition_size,
+    )
+
+    if use_coiled:
+        with coiled.Cluster(
+            n_workers=10,
+            worker_memory="4 GiB",
+            worker_options={"nthreads": 1},
+            region="us-east-2",
```

Comment:
If we're writing to a bucket we can infer what region that bucket is in to avoid cross region transfer charges.

Reply:
I think s3fs does not expose this but I haven't tried. Maybe https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem.info returns something useful?

Otherwise, with plain boto, see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/get_bucket_location.html#get-bucket-location

Reply:
sorted! https://github.com/coiled/benchmarks/pull/1131/commits/1e836fa504d7f130de3b41e1381e26a1f3fc16e3